x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheMaintenanceService.java

```diff
@@ -679,6 +689,16 @@ public void onFailure(Exception e) {
                 }
             }
         }
+
+        private void setCurrentResponse(SearchResponse response) {
```

Comment:
1. There's no massive changes in this method, I just extracted 2 methods to keep the indent level under control now that we need a loop to have correct thread-safety around the response reference.
2. I'm not sure this is the solution we want long-term. It might be better to just copy this kind of retained response to heap if it lives for an extended period of time compared to burdening the Netty buffer pool. Since it doesn't matter too much here, I decided to just handle this the straight forward way for now and not add a new API for that just yet.